### PR TITLE
docs: add redirect from updating-v10 to v11

### PR DESCRIPTION
**Pull request by kapunahelewong**
Add redirect from `updating-to-version-10.md` to v11

### CODE REVIEW

**Review by aikithoughts — APPROVED** [review]
Looks right to me!

Reviewed-for: global-docs-approvers

**Review by petebacondarwin — APPROVED** [review]
Could you add why this is needed to the commit message? Otherwise LGTM.

File: aio/firebase.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
       {"type": 301, "source": "/getting-started", "destination": "/start"},
       {"type": 301, "source": "/getting-started/:rest*", "destination": "/start/:rest*"},
       {"type": 301, "source": "/guide/displaying-data", "destination": "/start#template-syntax"},
+      {"type": 301, "source": "/guide/updating-to-version-10", "destination": "/guide/updating-to-version-11"},
```
Comment (gkalpak) [inline aio/firebase.json:35]: You also need to add corresponding entries in the [navigationUrls][1] list in `ngsw-config.json`.
(See [this commit][2].)

[1]: https://github.com/angular/angular/blob/a96afe9f324823064f42cf0f14a0db58ab3bf371/aio/ngsw-config.json#L71
[2]: https://github.com/angular/angular/blob/a96afe9f324823064f42cf0f14a0db58ab3bf371/aio/firebase.json#L8-L10

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/firebase.json (hunk 1 above)
Comment (petebacondarwin) [reply aio/firebase.json:35]: OMG! How did I forget that!

**Comment by aikidave** [comment]
Would it be possible to simplify our redirect system in some way? Both @kapunahelewong and I each struggle to remember all the places that we need to update. Perhaps we could have a single file that, along with a script, updates all the relevant redirect-related files? Just a thought.

**Comment by petebacondarwin** [comment]
To be fair there is a big comment at the top of this file: https://github.com/angular/angular/blob/a96afe9f324823064f42cf0f14a0db58ab3bf371/aio/firebase.json#L7-L11

**Comment by aikidave** [comment]
> To be fair there is a big comment at the top of this file:
> 
> https://github.com/angular/angular/blob/a96afe9f324823064f42cf0f14a0db58ab3bf371/aio/firebase.json#L7-L11

True, but if even you missed it, that's probably not a perfect solution. But perhaps we could at least expand the note so that it's understandable to a wider audience? Someone contributing to the documentation may not know what you mean by "service worker," nor what we mean when we say that it's "compatible" with the ngsw-config.json file. (What does "compatible" mean in this case?)

I realize this feature was never designed with non-engineers in mind. Yet, I think we agree that we'd all benefit from a more streamlined process?

**Comment by petebacondarwin** [comment]
The problem is that the regular expression in the service worker is not a straightforward mapping from this file. I suspect that someone very clever like @gkalpak might be able to come up with an algorithm to do the mapping but it wouldn't be trivial and there are probably corner cases. One option could be to keep both parts in a single file, as you suggest and then move them into the right files at deployment. But even this wouldn't be foolproof since you would need to know to fix both parts of the file.

**Comment by alxhub** [comment]
> To be fair there is a big comment at the top of this file:
> 
> https://github.com/angular/angular/blob/a96afe9f324823064f42cf0f14a0db58ab3bf371/aio/firebase.json#L7-L11

g3 has a nice CI system which allows for checking "if you change X, you should also change Y" style relationships. Perhaps we should build something similar for AIO? @josephperrott

**Comment by gkalpak** [comment]
Personally, I would like to make it more complicated to add redirects, so people are disincentivised to add more of them :grin:
<sub>(I am only partially kidding 😅)</sub>

**Comment by aikidave** [comment]
> > To be fair there is a big comment at the top of this file:
> > https://github.com/angular/angular/blob/a96afe9f324823064f42cf0f14a0db58ab3bf371/aio/firebase.json#L7-L11
> 
> g3 has a nice CI system which allows for checking "if you change X, you should also change Y" style relationships. Perhaps we should build something similar for AIO? @josephperrott

That would be such a great feature!

**Comment by aikidave** [comment]
> Personally, I would like to make it more complicated to add redirects, so people are disincentivised to add more of them 😁
> (I am only partially kidding 😅)

🤦

File: aio/firebase.json (hunk 1 above)
Comment (kapunahelewong) [reply aio/firebase.json:35]: Thank you both! I had read the comment in `firebase.json` and looked through `ngsw-config.json`, but was unsure. Is this right? https://github.com/angular/angular/pull/40108/files#diff-0352e74925a1c1e7c2cc38d9e6200d7ee43d45fc1ffa21bdef7718b305d3d66cR126

File: aio/ngsw-config.json [hunk 2]
```diff
@@ -123,6 +123,7 @@
     "!/guide/service-worker-configref",
     "!/guide/service-worker-configref.html",
     "!/guide/service-worker-configref/",
+    "!/guide/updating-to-version-11",
```
Comment (gkalpak) [inline aio/ngsw-config.json:126]:
```suggestion
    "!/guide/updating-to-version-10",
    "!/guide/updating-to-version-10.html",
    "!/guide/updating-to-version-10/",
```

**Review by gkalpak — CHANGES_REQUESTED** [review]
You could also add an entry in [aio/tests/deployment/shared/URLS_TO_REDIRECT.txt][1] to ensure the redirect is tested on CI:

```diff
 /guide/service-worker-configref	/guide/service-worker-config
+/guide//guide/updating-to-version-10	/guide//guide/updating-to-version-11
 /guide/webpack	https://v5.angular.io/guide/webpack
```

(Note that the two URLs must be separated by a tab character (not spaces).)

[1]: https://github.com/angular/angular/blob/e99d7aa469c10bff44bb1e88d3fcea5fde010e66/aio/tests/deployment/shared/URLS_TO_REDIRECT.txt

File: aio/firebase.json (hunk 1 above)
Comment (kapunahelewong) [reply aio/firebase.json:35]: Meant to tag you, @gkalpak! ✨

**Comment by kapunahelewong** [comment]
Thank you for your guidance, @gkalpak. 🌟 Ready for your re-review 🌸

**Comment by kapunahelewong** [comment]
PS: Hey @gkalpak, could you help me understand the failing test? A few minutes ago I fix the typo a few minutes ago but I don't know how to fix this failure:

```
Failures:
1) firebase.json redirect config with legacy urls should redirect the legacy urls
  Message:
    Expected '' not to equal ''.
```

😄

**Comment by petebacondarwin** [comment]
Did you inadvertently add an extra new line at the end of this file? https://github.com/angular/angular/pull/40108/files#diff-0b016061a23d83b3e711c4e0b8f0b18ba3d3624c55f6d2f45b618e3f9812282eR197

**Review by gkalpak — APPROVED** [review]
LGTM for docs-infra.

##
From a content perspective, though, I wonder whether a redirect to https://v10.angular.io/guide/updating-to-version-10 would be more appropriate.

A link to `updating-to-version-10` is more likely to be present in a context where an update specifically to v10 is being discussed, so a link to a guide describing how to update to the latest version (v11, v12, etc.) might be less relevant.

Leaving it up to you - I just wanted to bring it up.

**Comment by aikidave** [comment]
> LGTM for docs-infra.
> 
> From a content perspective, though, I wonder whether a redirect to https://v10.angular.io/guide/updating-to-version-10 would be more appropriate.
> 
> A link to `updating-to-version-10` is more likely to be present in a context where an update specifically to v10 is being discussed, so a link to a guide describing how to update to the latest version (v11, v12, etc.) might be less relevant.
> 
> Leaving it up to you - I just wanted to bring it up.

Good feedback, @gkalpak. Thank you! The docs team does need to discuss the most efficient way of redirecting this file as new releases come out. We'll look into that in a future PR.

**Pull request closed without merging**